after debug

HannaGo opened the pull request:
привіт!
допоможіть розібратися
я пробую сама дебажити код, але причесати до кінця не вдається. може я взагалі шось пропустила?

пробувала гуглити і знайшла що в подібному завданні використовують шось таке:
static {
        MERCURY.next = VENUS;
        VENUS.previous = MERCURY;
        VENUS.next = EARTH;
        EARTH.previous = VENUS;
        EARTH.next = MARS;
        MARS.previous = EARTH;
        MARS.next = JUPITER;
        JUPITER.previous = MARS;
        JUPITER.next = SATURN;
        SATURN.previous = JUPITER;
        SATURN.next = URANUS;
        URANUS.previous = SATURN;
        URANUS.next = NEPTUNE;
        NEPTUNE.previous = URANUS;
    }
    
    а потім надають змінним    previous і next тип Planet:
    ...
    private final double radius;
    **private Planet previous;
    private Planet next;**
    
    ми теж мали щось таке створити? я щось пропустила?
    чи можене допомогти і описати що за чим фіксати? пліііііз!

HannaGo committed:
after debug

LanaSvit commented:
Можна не ускладнювати, ви правильно зробили, що явно вказали для кожної планети її попередню. Якщо з наступною важче, можна пропустити, як домовлялися на лекції. Решта коментарів - в дз на платформі.

HannaGo committed:
It doesn't work when I'm diun calculation in constructor. it shows:
Caused by: java.lang.NullPointerException: Cannot invoke "[LPlanets;.clone()" because "Planets.$VALUES" is null

Why?

Without these calculations and without field distSun it works.
What is correct syntax for calculation in a constructor in enum class?
